lib/Tracing.js

```diff
@@ -69,20 +82,26 @@ class Tracing {
 
   /**
    * @param {string} handle
-   * @param {string} path
    */
-  async _readStream(handle, path) {
+  async _readResultFromStream(handle) {
     let eof = false;
-    const file = await openAsync(path, 'w');
+    this._result = '';
     while (!eof) {
       const response = await this._client.send('IO.read', {handle});
       eof = response.eof;
-      if (path)
-        await writeAsync(file, response.data);
+      this._result += response.data;
```

Comment:
This loads the whole trace into the node.js heap - do you really want to do this? Some traces could be huge and won't fit in.

Reply:
You're right, that's what I was thinking the other day..
Maybe the best would be just expose the stream, or convert it to a buffer and expose that.
What do you think?

Reply:
fwiw in node <= 8, the string length maximum was 256MB. you hit that pretty regularly with big comple traces.  in node 9 its 1gb, which is usually fine for traces.  https://stackoverflow.com/a/47781288/89484

so until puppeteer gave up support for node 8, we wouldnt be able to build a string like this. 

So yes, stream/buffer are other worthwhile options.

Reply:
> So yes, stream/buffer are other worthwhile options.

@paulirish @dios-david how would you process a trace that is saved into a buffer? You won't be able to `json.parse` it.